antarest/study/business/binding_constraint_management.py

```diff
@@ -30,21 +30,72 @@
 
 
 class LinkInfoDTO(BaseModel):
+    """
+    DTO for a constraint term on a link between two areas.
+
+    Attributes:
+        area1: the first area ID
+        area2: the second area ID
+    """
+
     area1: str
     area2: str
 
+    def calc_term_id(self) -> str:
+        """Return the constraint term ID for this link, of the form "area1%area2"."""
+        # Ensure IDs are in alphabetical order and lower case
+        ids = sorted((self.area1.lower(), self.area2.lower()))
+        return "%".join(ids)
+
 
 class ClusterInfoDTO(BaseModel):
+    """
+    DTO for a constraint term on a cluster in an area.
+
+    Attributes:
+        area: the area ID
+        cluster: the cluster ID
+    """
+
     area: str
     cluster: str
 
+    def calc_term_id(self) -> str:
+        """Return the constraint term ID for this Area/cluster constraint, of the form "area.cluster"."""
+        # Ensure IDs are in lower case
+        ids = [self.area.lower(), self.cluster.lower()]
```

Comment:
if I'm right, this will update the id in the constraint term config to be lowercase, from what you've told me, cluster ids shouldn't be lowercase.

This won't give any errors, because in the current tests it's not tested to modify the term of a BC containing a cluster, and the areas / cluster ids used are all in lowercase. (you can check the updated tests here: [#1903](https://github.com/AntaresSimulatorTeam/AntaREST/pull/1903), to include lower/upper test cases)

Reply:
# Identification des objets

## Cas des clusters thermiques

Dans la configuration d'une zone, les identifiants des clusters thermiques peuvent comporter des majuscules,
par exemple, dans le fichier `input/thermal/clusters/area2/list.ini`, on trouve dans la zone "Area1",
les clusters "Base", "Peek" et "SeminBase" :

```ini
[Base]
group = Nuclear
name = Base
co2 = 0.0

[Peak]
group = Oil
name = Peak
# ...

[SemiBase]
group = Hard coal
name = SemiBase
# ...
```

Cependant, lorsque les fichiers des chroniques sont écrits sur disque, on utilise des minuscules pour les dossiers
des clusters, par exemple :

```shell
$ find input/thermal/series/area2/ -name "series.txt"
input/thermal/series/area2/base/series.txt
input/thermal/series/area2/peak/series.txt
input/thermal/series/area2/semibase/series.txt
```

## Cas des contraintes couplantes

### Contrainte couplante
Lorsqu'on crée une contrainte couplante, le nom peut comporter des majuscules, mais l'ID est en minuscules.
Il n'y a pas de problème à ce niveau.

Par exemple, si je crée une contrainte nommée "New CC", on a bien l'identifiant "new cc" dans
le fichier `input/bindingconstraints/bindingconstraints.ini` :

```ini
[0]
name = Storing_balance
id = storing_balance
# ...

[1]
name = New CC
id = new cc
enabled = True
type = daily
operator = equal
comments = Nouvelle contrainte couplante
```

Il en est de même lorsqu'on écrit la chronique de la contrainte couplante dans un fichier :
il faut utiliser l'identifiant en minuscules. Par exemple `new cc.txt` :

```shell
$ find input/bindingconstraints -name *.txt
input/bindingconstraints/new cc.txt
input/bindingconstraints/storing_balance.txt
```

### Terme d'une contrainte couplante

Lorsqu'on ajoute un terme à cette contrainte afin qu'elle soit calculée en fonction des paramètres de production
d'un cluster thermique, on doit _afficher_ le **nom** de la zone et le **nom** d'un cluster dans cette zone.
Ces noms peuvent contenir des majuscules.

![image](https://github.com/AntaresSimulatorTeam/AntaREST/assets/43534797/4d31e3ab-e714-4fdc-bc18-7c1055e0fa53)

Mais lorsqu'on _sauvegarde_ les données dans le fichier de configuration, on doit utiliser les identifiants
et les convertir en **minuscules**.
Par exemple, avec le cluster "Base" de la zone "Area1", on doit afficher "Area1" et "Base",
mais enregistrer l'identifiant du terme en minuscules, sous la forme "area1.base", comme ci-dessous :

```ini
[0]
name = Storing_balance
id = storing_balance
# ...

[1]
name = New CC
id = new cc
enabled = True
type = daily
operator = equal
comments = Nouvelle contrainte couplante
area2.base = 0.0
```